### DISCUSSION TITLE
2021 Q3 Roadmap 🎯

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#61] [News]
As you may know, we have been working on the new version 0.21 of the engine for quite some time. We have started to iterate on a release candidate that you can now test. We have made a [blog post](https://blog.meilisearch.com/refactoring-meilisearch/) about it if you want to know more.

It's time to think about the future and move forward. That's why I'd like to give you more visibility on what we have planned for the third quarter of 2021. 

We have set some clear goals for Q3 🎯. 

Our first goal is to reduce what some of you have called deal-breakers when it comes to choosing MeiliSearch. We will therefore devote a maximum of our efforts to these two subjects in order to improve the search experience for you and your users by being able to manage more use-cases with MeiliSearch.

Our second goal is to start building a guide-line to facilitate future developments and upgrades of the API. This goal will also allow us to improve the developer experience to make it easier for you to use the API. We will tackle this problem block by block to move towards a 1.0 stable version.

Finally, we are aware that there are still things to improve to reduce indexing times. We have decided to take advantage of Q3 to iterate on this subject directly after the release of v0.21. In order to accomplish this objective, we will also provide two new formats to index your documents in MeiliSearch. We will add the support of CSV and NDJSON files for indexing.

If we summarize what we have just seen, here is a list of objectives and associated steps.

## Reduce deal-breakers
 - Add SortBy feature
 - Add Geo-search feature
 
## Stabilize the API format block by block until v1.0.
- Write an API specification guide-line.
    - Search endpoints
    - Documents endpoints
    - Updates endpoints
    - Indexes endpoints

## Patch Indexing speed
- Add NDJSON file format
- Add CSV file format
- Optimize the indexer

You can visualize it on the dedicated [Public Core Roadmap](https://github.com/meilisearch/product/projects/5) project page.

Thank you for your reading ✌️

**Comment by bb** [comment]
Is https://github.com/meilisearch/MeiliSearch/issues/1436 also planned for Q3?

**Reply by gmourier** [reply]
Hello @bb! 👋

It is not planned for Q3, we will go back over all the resources mentioned here to clarify and bring them in line with our API convention that we are working on.

At this stage, I think that only the settings and search resources will receive the necessary additions to make the geo-search and sortBy features work.
